.travis.yml

```diff
@@ -6,12 +21,22 @@ os:
 jdk:
   - oraclejdk8
 
+# Caches mvn repository in order to speed upbuilds
+cache:
+  directories:
+  - $HOME/.m2
+  - $HOME/.npm
+
 # before_install aids in a couple workarounds for issues within the Travis-CI environment
 #   1. Workaround for buffer overflow issues with OpenJDK versions of java as per https://github.com/travis-ci/travis-ci/issues/5227#issuecomment-165135711
 #   2. Workaround for non-existent Maven repository as per: https://github.com/travis-ci/travis-ci/issues/4629#issuecomment-131214465
 before_install:
   - sudo hostname "$(hostname | cut -c1-63)"
   - sed -e "s/^\\(127\\.0\\.0\\.1.*\\)/\\1 $(hostname | cut -c1-63)/" /etc/hosts | sudo tee /etc/hosts
   - sed -i.bak -e 's|https://nexus.codehaus.org/snapshots/|https://oss.sonatype.org/content/repositories/codehaus-snapshots/|g' ~/.m2/settings.xml
+  - echo "MAVEN_OPTS='-Xmx2g -XX:+UseG1GC'" > ~/.mavenrc
+  - mvn -Pcontrib-check apache-rat:check checkstyle:check
```

Comment:
@apiri 

Yeah... that was a bit weird but it was an attempted workaround run checkstyle:check before compilation time. 

Sadly it didn't work, when we run checkstyle check, it triggers some style errors that have been disabled in our style rules:

`[WARNING] src/main/java/org/apache/nifi/reporting/ReportingTask.java[64:8] (javadoc) JavadocMethod: Unable to get class information for @throws tag 'org.apache.nifi.reporting.InitializationException'.`